src/operations/blas3/gemm_local_tensorcore.hpp

```diff
@@ -0,0 +1,814 @@
+  static constexpr index_t wg_size = wg_rows * wg_cols;
+  //! @brief Number of rows within a work-group level tile
+  static constexpr index_t block_rows = wg_rows * item_rows;
+  //! @brief Number of columns within a work-group level tile
+  static constexpr index_t block_cols = wg_cols * item_cols;
+  //! @brief Number of rows within a top-level tile
+  static constexpr index_t big_tile_rows = tl_rows * block_rows;
+  //! @brief Number of columns within a top-level tile
+  static constexpr index_t big_tile_cols = tl_cols * block_cols;
+
+  static constexpr index_t jm_row_frags = block_rows / tile_type::sg_M;
+  static constexpr index_t jm_col_frags = block_cols / tile_type::sg_N;
+  static constexpr index_t num_jm_frags = jm_row_frags * jm_col_frags;
+  static constexpr index_t num_sub_groups = wg_size / sg_size;
+  static constexpr index_t frags_per_sg = num_jm_frags / num_sub_groups;
+
+  static_assert(wg_size % cl_elems == 0,
+                "Work group size should be a multiple "
+                "of elements in a cache line\n"
+                " --- this is ensured iff:"
+                " cl_size | sizeof(element_t) * wg_rows * wg_cols");
+
+  static_assert(wg_size % block_rows == 0,
+                "Work group size should be a multiple "
+                "of the number of rows in a block\n"
+                " --- this is ensured iff: item_rows | wg_cols");
+
+  static_assert(wg_size % block_cols == 0,
+                "Work group size should be a multiple "
+                "of the number of columns in a block\n"
+                " --- this is ensured iff: item_cols | wg_rows");
+
+  static_assert(big_tile_rows == big_tile_cols,
+                "Big tile level dimensions should be square, i.e. tl_rows * "
+                "block_rows == tl_cols * block_cols");
+
+  static_assert(item_rows % packetize_t::packet_size == 0,
+                "Item rows must be a multiple of the vector packet size");
+
+  static_assert(cl_elems % packetize_t::packet_size == 0,
+                "Cache line size must be a multiple of packet_size");
+
+  //! @brief leading dimension of block of A in local
+  static constexpr index_t ldsa = block_rows + nbc_a;
+  //! @brief leading dimension of block of B in local
+  static constexpr index_t ldsb = cl_elems + nbc_b;
+  //! @brief size (in elements) of local (local) memory required by each
+  //         work group
+  static constexpr index_t local_memory_size =
+      (double_buffer + 1) * (ldsa * cl_elems + ldsb * block_cols);
+
+  input_t a_;
+  input_t b_;
+  output_t c_;
+  const element_t alpha_;
+  const element_t beta_;
+  index_t batch_size_;
+
+  SYCL_BLAS_INLINE Gemm(input_t A, input_t B, output_t C, element_t alpha,
+                        element_t beta, index_t batch_size)
+      : a_(A),
+        b_(B),
+        c_(C),
+        alpha_(alpha),
+        beta_(beta),
+        batch_size_(batch_size) {}
+
+  /*!
+   * @brief Get the type of this GemmFactory as a human readable string.
+   */
```

Comment:
Can some of these docs just be `/// Get the type of this GemmFactory as a human readable string.`? It's the same [to doxygen](https://doxygen.nl/manual/docblocks.html).

Reply:
This way of adding documentation is actually copied from the `src/operations/blas3/gemm_local.hpp` file. I think it is the same in all other `gemm_*` files in this directory.